xmake/rules/c++/modules/modules_support/clang/dependency_scanner.lua

```diff
@@ -0,0 +1,83 @@
+--!A cross-platform build utility based on Lua
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+-- Copyright (C) 2015-present, TBOOX Open Source Group.
+--
+-- @author      ruki, Arthapz
+-- @file        clang/dependency_scanner.lua
+--
+
+-- imports
+import("core.base.json")
+import("core.base.semver")
+import("core.project.depend")
+import("utils.progress")
+import("compiler_support")
+import(".dependency_scanner", {inherit = true})
+
+-- generate dependency files
+function generate_dependencies(target, sourcebatch, opt)
+    local compinst = target:compiler("cxx")
+    local changed = false
+    for _, sourcefile in ipairs(sourcebatch.sourcefiles) do
+        local dependfile = target:dependfile(sourcefile)
+        depend.on_changed(function()
+            if opt.progress then
+                progress.show(opt.progress, "${color.build.target}<%s> generating.module.deps %s", target:name(), sourcefile)
+            end
+
+            local outputdir = compiler_support.get_outputdir(target, sourcefile)
+            local jsonfile = path.translate(path.join(outputdir, path.filename(sourcefile) .. ".json"))
+            if compiler_support.has_clangscandepssupport(target) and not target:policy("build.c++.clang.fallbackscanner") then
+                -- We need absolute path of clang to use clang-scan-deps
+                -- See https://clang.llvm.org/docs/StandardCPlusPlusModules.html#possible-issues-failed-to-find-system-headers
+                local clang_path = compinst:program()
+                if not path.is_absolute(clang_path) then
+                    clang_path = compiler_support.get_clang_path(target) or compinst:program()
+                end
+                local clangscandeps = compiler_support.get_clang_scan_deps(target)
+                local compflags = compinst:compflags({sourcefile = sourcefile, target = target})
+                local flags = table.join({"--format=p1689", "--",
+                                         clang_path, "-x", "c++", "-c", sourcefile, "-o", target:objectfile(sourcefile)}, compflags or {})
+                vprint(table.concat(table.join(clangscandeps, flags), " "))
```

Comment:
This will cause table.concat to be executed every time, even if verbose is not enabled, and please use os.args instead of table.concat

```lua
import("core.base.option")
if option.get("verbose") then
    print(os.args(table.join(clangscandeps, flags)))
end
```

Reply:
and why not use `os.vrunv`?

Reply:
and here